### DISCUSSION TITLE
MQTT JSON benutzen

### DISCUSSION THREAD

**Discussion by paddy-ol** [Zefau/ioBroker.jarvis#1755] [General]
Hallo,
ich bin noch neu in jarvis, also verzeiht meine Frage. 
Ich habe mehrere Steckdosenadapter mit Tasmota laufen. Diese Funken per MQTT an iobroker.
Dabei gibt es z.B. für die aktuelle Leistung ein Statusobjekt, das von Tasmota geschickt wird.
Kann ich das irgendwie in jarvis benutzen, um an `Power` zu kommen oder muss ich das per Skript oder NodeRed aufdröseln in ein separates Nummernobjekt?

Beispiel:
`{
  "StatusSNS": {
    "Time": "2022-06-27T19:06:04",
    "ENERGY": {
      "TotalStartTime": "2022-05-08T09:49:23",
      "Total": 7.577,
      "Yesterday": 0.56,
      "Today": 0.215,
      "Power": 27,
      "ApparentPower": 54,
      "ReactivePower": 47,
      "Factor": 0.5,
      "Voltage": 232,
      "Current": 0.233
    }
  }
}`

Zweite Frage:
Wenn ich sowas als Slider darstelle in jarvis, wie kann ich das als "read only" markieren, damit der Slider nicht schreibt auf das Objekt?

**Comment by mcuiobroker** [comment]
1. https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/geraet/datenpunkt-eigenschaft/json-parse
2. Datenpunkt-Trigger entfernen 
![image](https://user-images.githubusercontent.com/84818934/176009303-eb4dfa47-c192-4ab7-baf7-6f559c143c65.png)

**Reply by Zefau** [reply]
read-only geht mit v3 nicht mehr, da ein leerer Trigger immer den State anspricht als Fallback. Müsste ich eine Option einbauen.

**Reply by paddy-ol** [reply]
Genau das hatte ich gemerkt mit Temperaturwerten von Zigbee Sensoren. Die habe ich mit dem Slider dargestellt, um einfach zu sehen wie die Aussentemperatur im Vergleich zum z.B. Schlafzimmer ist und Lüften angebracht ist oder besser zu lassen. Aber mit dem Smartphone konnte man die Werte einfach verstellen (hatte kein Datenpunkt-Trigger eingestellt). Die Werte blieben dann bei allen Teilnehmern verstellt, bis vom Sensor eine neue Messung kam.

@Zefau wäre super, wenn du das einbauen kannst.

**Reply by mcuiobroker** [reply]
Eine Option im LevelBody "read-Only" würde ja dann reichen? Dann kann man ihn nicht mehr verstellen.

**Reply by paddy-ol** [reply]
Danke für den Link zur Doku. Ich war in der v2 Doku unterwegs, weil in der v3 nur die leere "initial Page" ist. Vielleicht könnte da wenigstens ein Link zu jarvis-infos rein.

Ob die Option "read-only" in den LevelBody als View gehört oder besser in den Datenpunkt kann ich noch nicht überblicken. Das muss der Profi entscheiden. "Reichen" würde es erst mal im LevelBody.

**Comment by paddy-ol** [comment]
Sehe ich das richtig, dass die Option mit json-parse in der PRO steckt?
Solange ich keine PRO habe müsste ich das "von Hand" in iobroker zerlegen?

**Reply by mcuiobroker** [reply]
Ja

**Comment by Zefau** [comment]
https://github.com/Zefau/ioBroker.jarvis/issues/1756
